doc/source/reference/c-api/iterator.rst

```diff
@@ -203,6 +203,67 @@ is used to control the memory layout of the allocated result, typically
     }
 
 
+Multi Index Tracking Example
+----------------------------
+
+This example shows you how to work with the :c:data:`NPY_ITER_MULTI_INDEX` flag. For simplicity, we assume the argument is a two-dimensional array.
+
+.. code-block:: c
+
+   int PrintMultiIndex(PyArrayObject *arr) {
+       NpyIter *iter;
+       NpyIter_IterNextFunc *iternext;
+       npy_intp *multi_index;
+
+       iter = NpyIter_New(
+           arr, NPY_ITER_READONLY | NPY_ITER_MULTI_INDEX | NPY_ITER_REFS_OK,
+           NPY_KEEPORDER, NPY_NO_CASTING, NULL);
+       if (iter == NULL) {
+           return -1;
+       }
+       if (NpyIter_GetNDim(iter) != 2) {
+           NpyIter_Deallocate(iter);
+           PyErr_SetString(PyExc_ValueError, "Array must be 2-D");
+           return -1;
+       }
+       if (NpyIter_GetIterSize(iter) != 0) {
+           iternext = NpyIter_GetIterNext(iter, NULL);
+           if (iternext == NULL) {
+               NpyIter_Deallocate(iter);
+               return -1;
+           }
+           NpyIter_GetMultiIndexFunc *get_multi_index =
+               NpyIter_GetGetMultiIndex(iter, NULL);
+           if (get_multi_index == NULL) {
+               NpyIter_Deallocate(iter);
+               return -1;
+           }
+
+           multi_index = (npy_intp *)PyArray_DATA(arr);
```

Comment:
```suggestion
```

Reply:
You would be writing the index into the array passed in, which would be wrong.  I do wonder if we want to add a comment in the inner loop saying something like:
```
/*
 * Can use array values via NpyIter_GetDataPtrArray. 
 */
```
But after fixing the multi-index storage, I think this is good to go.  Thanks.